File: 404-sum-of-left-leaves/sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        isLeft = False

        def leftSum(node: Optional[TreeNode], isLeft: bool) -> int:
            if node is None:
                return 0
            if node.left is None and node.right is None:
                if isLeft:
                    return node.val
                else:
                    return 0

            return leftSum(node.left, True) + leftSum(node.right, False)

        return leftSum(root, isLeft)

File: 404-sum-of-left-leaves/sum-of-left-leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        if (root == null) return 0;

        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);

        int ans = 0;

        while(!queue.isEmpty()) {
            TreeNode node = queue.poll();

            if (node.left != null) {
                if (node.left.left == null && node.left.right == null) {
                    ans += node.left.val;
                } else {
                    queue.offer(node.left);
                }
            }

            if (node.right != null) {
                queue.offer(node.right);
            }
        }

        return ans;
    }
}